### ISSUE TITLE
Bot scanner for bump version

### ISSUE THREAD

**Issue by uilianries** [bincrafters/community#185]
Hi!

For now we need to check each project release and create a new issue to bump the version.

It would be nice if we would have some bot to check every day all projects and open the issue if some version needs to be updated.

I though about a simple script running over Travis cron, but I don't know if someone already did this.

**Comment by Croydon** [comment]
This should be fairly easy for projects who are using git tags or use some kind of release RSS. Many projects only change a version string somewhere in the source code. It would be rather hard to find a generic solucation for these kind of projects.

**Comment by uilianries** [comment]
I agree. My first intention was create a bot to open issue and tag as "update upstream".

@solvingj has a better idea using Github's marketplace.

**Comment by theirix** [comment]
I use RSS for releases for packaging I am looking after. For a typical GH repository it is something like https://github.com/open-source-parsers/jsoncpp/releases.atom . So we can write a simple RSS aggregator that tracks versions. For other packages with development outside GH (about 50%) the problem is more complicated, versions are announced via webpage or mailing list.

There were some services like allmychanges but none of them works well now. They gathered information from changelogs too.

In debian packages, we are using [watch](https://wiki.debian.org/debian/watch) - a simple shell script that is called by debian infrastructure that returns a current upstream version. We can grab a lot of working urls because debian packages are very good at maintaining or just hook up these watches from apt sources.

**Comment by Croydon** [comment]
>  For a typical GH repository it is something like https://github.com/open-source-parsers/jsoncpp/releases.atom . So we can write a simple RSS aggregator that tracks versions.

As mentioned earlier this only works when the projects are using GitHub release or at least git tags, but not all (GitHub) projects are doing that.

**Comment by uilianries** [comment]
I found the project https://coderelease.io

It sends email when a new release is created. But only works for github projects. Fortunately, almost all projects that we maintain are present on Github, or there is a mirror.

We could use coderelease project to trigger a service that bump the recipe version.

However, we will need a non-free service to do this. An email or even RSS needs a service to start a new function and I don't know about a free container service. We could use some trial services provided by Amazon, Azure, Google ... but I'll migrate after some while.

**Comment by uilianries** [comment]
@SSE4 found many other projects:

https://libraries.io/
https://touchpine.com/
https://releaser.richardhagen.no/
https://gitpunch.com/
https://coderelease.io/
https://rping.github.io/Ver.bot-site/
https://ossindex.net/
